app/javascript/src/QuickStarts/templates/basicAPIintegration.yaml

```diff
@@ -1,9 +1,8 @@
 # Additional info: https://docs.openshift.com/container-platform/4.9/web_console/creating-quick-start-tutorials.html
 metadata:
   name: getting-started-with-quick-starts
-  # you can add additional metadata here
-  instructional: true
-  learningResources: true
+  labels:
+    category: basic-api-integration-setup
```

Comment:
I find this structure not much intuitive. Why is `category` under `labels`? If you are referring to it as `category.id` from React why not using the same naming here?

```yaml
category:
    id: basic-api-integration-setup
```

or:

```yaml
category_id: basic-api-integration-setup
```

Reply:
> Why is `category` under `labels`?

We use type `QuickStart` from `@patternfly/quickstarts`, see below how it is defined.
While we could just extend the type and add `category` field to the metadata, I don't think it's worth it. I think putting **labels** to the object metadata is a common way to mark objects in a certain way, and it's already a freeform field, we could add any more labels later and avoid adding custom types.

Now, specifying it as `categoryId` could be fine, but I don't think it's necessary either :shrug: I think by looking at the yaml, it's very clear that this label just defines the category that the QuickStart object belongs to (it could be seen as an "ID", as a "name", or whatever else).

Now, in our own code **internally** we are referencing this value as an **ID** of the category, just because we cant to map the more machine-looking name to a "nicer looking" title.

We could get rid of it altogether, and just include the full title in the yaml, e.g.

```
   metadata:
     name: getting-started-with-quick-starts
     category: 'Common use cases for creating and tracking APIs'
```
But I think it's not cool.

```
export declare type QuickStart = {
    apiVersion?: string;
    kind?: string;
    metadata: ObjectMetadata;
    spec: QuickStartSpec;
};

export declare type ObjectMetadata = {
    annotations?: {
        [key: string]: string;
    };
    clusterName?: string;
    creationTimestamp?: string;
    deletionGracePeriodSeconds?: number;
    deletionTimestamp?: string;
    finalizers?: string[];
    generateName?: string;
    generation?: number;
    labels?: {
        [key: string]: string;
    };
    managedFields?: any[];
    name?: string;
    namespace?: string;
    ownerReferences?: OwnerReference[];
    resourceVersion?: string;
    uid?: string;
};
```

Reply:
Ok, we are fitting the data into a predefined structure.